lib/AST/FrontendSourceFileDepGraphFactory.cpp

```diff
@@ -92,24 +93,29 @@ DependencyKey::Builder DependencyKey::Builder::fromReference(
   case Kind::Tombstone:
     llvm_unreachable("Cannot enumerate dead reference!");
   case Kind::PotentialMember:
-    return Builder{kind, aspect}.withContext(ref.subject);
+    return Builder{kind, aspect}.withContext(ref.subject->getAsDecl());
   case Kind::TopLevel:
   case Kind::Dynamic:
     return Builder{kind, aspect, nullptr, ref.name.userFacingName()};
   case Kind::UsedMember:
     return Builder{kind, aspect, nullptr, ref.name.userFacingName()}
-        .withContext(ref.subject);
+        .withContext(ref.subject->getAsDecl());
   }
 }
 
 DependencyKey::Builder DependencyKey::Builder::withContext(const Decl *D) && {
   switch (kind) {
   case NodeKind::nominal:
   case NodeKind::potentialMember:
-  case NodeKind::member:
+  case NodeKind::member: {
     /// nominal and potential member dependencies are created from a Decl and
     /// use the context field.
-    return Builder{kind, aspect, cast<NominalTypeDecl>(D), name};
+    const DeclContext *context = dyn_cast<NominalTypeDecl>(D);
+    if (!context) {
+      context = cast<ExtensionDecl>(D);
```

Comment:
Not new with this PR, and a nit: When I saw the "cast" I realized that `withContext` will fail an assert when the `D` parameter is the wrong kind of `Decl`. Perhaps the declaration of `withContext` in the header file would benefit from a comment stating the required types of that parameter?